From 96be998ab95cdc9467f382f47d2b93854ecb6ecf Mon Sep 17 00:00:00 2001
From: Satish Kumar Nagireddy <satish.nagireddy.nagireddy@xilinx.com>
Date: Wed, 24 Jul 2019 19:33:54 -0700
Subject: [PATCH 0646/1851] v4l: xilinx: dma: Check subdev resolution against
 crop rectangle

commit 1619d84e041c5d837d1e7a1dcc7b4737ffd208a7 from
https://github.com/Xilinx/linux-xlnx.git

The crop rectangle contains the format resolution by default and
selected resolution if s_selection is executed. So this patch
verifies the subdev resolution against crop rectangle.

Signed-off-by: Satish Kumar Nagireddy <satish.nagireddy.nagireddy@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/media/platform/xilinx/xilinx-dma.c | 16 ++++++----------
 1 file changed, 6 insertions(+), 10 deletions(-)

diff --git a/drivers/media/platform/xilinx/xilinx-dma.c b/drivers/media/platform/xilinx/xilinx-dma.c
index 694dd9644de7..7632bb62a4e4 100644
--- a/drivers/media/platform/xilinx/xilinx-dma.c
+++ b/drivers/media/platform/xilinx/xilinx-dma.c
@@ -67,7 +67,6 @@ static int xvip_dma_verify_format(struct xvip_dma *dma)
 	struct v4l2_subdev_format fmt;
 	struct v4l2_subdev *subdev;
 	int ret;
-	int width, height;
 
 	subdev = xvip_dma_remote_subdev(&dma->pad, &fmt.pad);
 	if (!subdev)
@@ -81,15 +80,12 @@ static int xvip_dma_verify_format(struct xvip_dma *dma)
 	if (dma->fmtinfo->code != fmt.format.code)
 		return -EINVAL;
 
-	if (V4L2_TYPE_IS_MULTIPLANAR(dma->format.type)) {
-		width = dma->format.fmt.pix_mp.width;
-		height = dma->format.fmt.pix_mp.height;
-	} else {
-		width = dma->format.fmt.pix.width;
-		height = dma->format.fmt.pix.height;
-	}
-
-	if (width != fmt.format.width || height != fmt.format.height)
+	/*
+	 * Crop rectangle contains format resolution by default, and crop
+	 * rectangle if s_selection is executed.
+	 */
+	if (dma->r.width != fmt.format.width ||
+	    dma->r.height != fmt.format.height)
 		return -EINVAL;
 
 	return 0;
-- 
2.31.1

